gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +721,671 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
+                oSRS.SetProjection(SRS_PT_LAMBERT_CONFORMAL_CONIC_1SP);
+                oSRS.SetProjParm(SRS_PP_LONGITUDE_OF_CENTER, crval1);
+                /*
+            #clat = self.__header['XXXXX']
+            #srs.SetProjParm('latitude_of_center',clat)
+            scale = self.__header['XXXXX']
+            if scale is not None:
+                srs.SetProjParm('scale_factor',scale)
+            else: #set default of 1.0
+                srs.SetProjParm('scale_factor',1.0)
+                */
+
+            /* Orthographic / SIN projection */
+            } else if( strcmp(projstr,"SIN" ) == 0 ) {
+                strncpy(projName, "Orthographic_", 14);
+                oSRS.SetProjection(SRS_PT_ORTHOGRAPHIC);
+                oSRS.SetProjParm(SRS_PP_CENTRAL_MERIDIAN, crval1);
+                /*
+            #olat = self.__header['XXXXX']
+            #srs.SetProjParm('latitude_of_origin',olat)
+                */
+
+            /* Point Perspective / AZP projection */
+            } else if( strcmp(projstr,"AZP" ) == 0 ) {
+                strncpy(projName, "perspective_point_height_", 26);
+                oSRS.SetProjection(SRS_PP_PERSPECTIVE_POINT_HEIGHT);
+            /* # appears to need height... maybe center lon/lat */
+
+            /* Polar Stereographic / STG projection */
+            } else if( strcmp(projstr,"STG" ) == 0 ) {
+                strncpy(projName, "Polar_Stereographic_", 21);
+                oSRS.SetProjection(SRS_PT_POLAR_STEREOGRAPHIC);
+                oSRS.SetProjParm(SRS_PP_CENTRAL_MERIDIAN, crval1);
+                oSRS.SetProjParm(SRS_PP_LATITUDE_OF_ORIGIN, crval2);
+            } else {
+                CPLError(CE_Failure, CPLE_AppDefined,
+                     "Unknown projection.");
+            }
+        }
+
+        strcat(projName, target);
+        oSRS.SetProjParm(SRS_PP_FALSE_EASTING,0.0);
+        oSRS.SetProjParm(SRS_PP_FALSE_NORTHING,0.0);
+        oSRS.SetNode("PROJCS",projName);
+        oSRS.SetGeogCS(pszGeogName, pszDatumName, target, aRadius, invFlattening,
+            "Reference_Meridian", 0.0, "degree", 0.0174532925199433);
+
+    }
+
+    if( m_bLoadPam )
+    {
+/* -------------------------------------------------------------------- */
+/*      Initialize any PAM information.                                 */
+/* -------------------------------------------------------------------- */
+        CPLAssert(!bMetadataChanged);
+        CPLAssert(!bNoDataChanged);
+
+        m_bLoadPam = false;
+        bMetadataChanged = false;
+        bNoDataChanged = false;
+
+        bitpix = this->fitsDataType;
+        FITSRasterBand *poBand = cpl::down_cast<FITSRasterBand*>(GetRasterBand(1));
+
+        if (bitpix != TUSHORT && bitpix != TUINT)
+        {
+            fits_read_key(hFITS, TDOUBLE, "BSCALE", &dfScale, nullptr, &status);
+            if( status )
+            {
+                status = 0;
+                dfScale = 1.;
+            }
```

Comment:
You're setting status but you're going to overwrite it in the next call, no?

Reply:
I reinitialize `status` to 0 after a possible CFITSIO error status output. Some keywords are not mandatory: if they are not there I set the corresponding variable to the default FITS standard value